admin/app/controllers/workarea/admin/custom_events_controller.rb

```diff
@@ -0,0 +1,38 @@
+module Workarea
+  module Admin
+    class CustomEventsController < Admin::ApplicationController
+      before_action :find_event, except: :create
```

Comment:
So the `@custom_event` is only used in `update` so this doesn't really make sense to run before every action in its current state. You could either separate this out and not use the `before_action` or, rewrite `find_event` to do something that works for all actions, like
```
def find_event
  @event = if params[:id].present?
    Reports::CustomEvent.find(params[:id])
  else
    Reports::CustomEvent.new(params[:event])
  end
end
```

Reply:
edit - maybe it was needed was for views? In which case, you could just do the above and wrap the result in the view model.